src/UCTNode.cpp

```diff
@@ -47,36 +47,26 @@ bool UCTNode::first_visit() const {
     return m_visits == 0;
 }
 
-SMP::Mutex& UCTNode::get_mutex() {
-    return m_nodemutex;
-}
-
 bool UCTNode::create_children(Network & network,
                               std::atomic<int>& nodecount,
                               GameState& state,
                               float& eval,
                               float min_psa_ratio) {
-    // check whether somebody beat us to it (atomic)
-    if (!expandable(min_psa_ratio)) {
-        return false;
-    }
-    // acquire the lock
-    LOCK(get_mutex(), lock);
     // no successors in final state
     if (state.get_passes() >= 2) {
         return false;
     }
-    // check whether somebody beat us to it (after taking the lock)
-    if (!expandable(min_psa_ratio)) {
+
+    // acquire the lock
+    if (!acquire_expanding()) {
         return false;
     }
-    // Someone else is running the expansion
-    if (m_is_expanding) {
+
+    // can we actually expand?
+    if (!expandable(min_psa_ratio)) {
```

Comment:
You introduced a bug now: if you choose not to expand a node because it is already expanded up to the appropriate min_psa_ratio, you will leave the node in INITIAL state, but it should really be left to EXPANDED.